### PR TITLE
Initial support for multiple aurora versions

### PR DESCRIPTION
**Pull request by mateuszchyla**
This reuses checksum calc logic from the loaded, but doesn't have registry.

I have tested it with 3 Aurora.exe versions - AuroraMod, Vanilla Aurora, and Cleaned Aurora (used de4dot)

### CODE REVIEW

File: AuroraPatch/AuroraPatch.csproj [hunk 1]
```diff
@@ -32,6 +32,10 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed">
+      <HintPath>..\packages\Newtonsoft.Json.12.0.3\lib\net40\Newtonsoft.Json.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
```
Comment (mateuszchyla) [inline AuroraPatch/AuroraPatch.csproj:38]: I wanted to use `System.Text.Json`, but looks like it is not supported with .NET Framework v4.0

**Comment by 01010100b** [comment]
The patcher dynamically identifies the TacticalMap through a heuristic (between 60 and 80 buttons and checkboxes) so as to be broadly compatible with Aurora versions as well as exe mods. Version-specific mapping should happen downstream in the patches themselves. Patches will be "mods" in the sense as used by AuroraLoader and hence versioned against Aurora versions, the patcher itself will be part of the loader itself and hence broadly compatible with all Aurora versions and exe mods.

**Comment by mateuszchyla** [comment]
@01010100b Yep, that's fine. The only reason I wanted to do it was because counting checkboxes and buttons seems like a bad idea, but I do agree with your reasoning.

**Comment by 01010100b** [comment]
Yes I was going to revisit that heuristic later (run some experiments to see what good heuristics are) but it works for now so that's not a priority.

**Pull request closed without merging**